cli/azd/cmd/root.go

```diff
@@ -106,7 +106,7 @@ For more information, visit the Azure Developer CLI Dev Hub: https://aka.ms/azur
 	cmd.AddCommand(telemetryCmd(opts))
 	cmd.AddCommand(templatesCmd(opts))
 
-	cmd.AddCommand(BuildCmd(opts, versionCmdDesign, initVersionAction, &buildOptions{disableTelemetry: true}))
+	cmd.AddCommand(BuildCmd(opts, versionCmdDesign, initVersionAction, &actions.ActionOptions{DisableTelemetry: true}))
```

Comment:
I wonder if instead of enabling the telemetry middleware, then disabling it via option, we could instead define the middleware pipeline per CLI command, and just omit the telemetry middleware when we don't want it configured

Reply:
We also avoid leaking "DisableTelemetry" to all middleware components